Sources/WFOAuth2/WFOAuth2SessionManager.m

```diff
@@ -256,8 +256,15 @@ - (WKWebView *)authorizationWebViewWithURL:(NSURL *)authorizationURL
         if ([redirectComponents.path hasSuffix:@"/"])
             redirectComponents.path = [redirectComponents.path substringToIndex:(redirectComponents.path.length - 1)];
         
+        /*
+            Square requires you to save the redirect URL in the webportal, there is no sense in passing it in as a parameter.
+            The assumption is made that redirecting to something other than the square domain is considered redirecting to your redirect url.
+         */
+        if (!redirectURI && [components.host containsString:@"squareup"])
```

Comment:
Yes, I was hesitant with this section as well. I should have cleaned it up from the start, sorry. I like your idea of scanning for completion parameters. As I understand the OAuth 2.0 spec, the required parameters in our case would be `code` and `state`. In the case of an error the sole required parameter would be `error`. I'll scan for those in the case of a `nil` `redirectURI`. I'll update the branch shortly.